### PR TITLE
Ignore self-loops when resolving symlinks.

### PR DESCRIPTION
**Pull request by anakrish**
This prevents infinite recursion.

Signed-off-by: Anand Krishnamoorthi <anakrish@microsoft.com>

### CODE REVIEW

File: ext2/ext2.c [hunk 1]
```diff
@@ -1334,15 +1334,20 @@ static int _path_to_ino_recursive(
...
+                    ECHECK(_path_to_ino_recursive(
+                        ext2,
+                        locals->target,
+                        current_ino,
+                        FOLLOW,
+                        &current_ino,
+                        &locals->ino,
+                        realpath,
+                        target_out));
+                }
```
Comment (vtikoo) [inline ext2/ext2.c:1350]: Does this return an error code if its a self-loop?

**Review by vtikoo — COMMENTED** [review]
Similar patch might work for ramfs as well.

File: ext2/ext2.c (hunk 1 above)
Comment (anakrish) [reply ext2/ext2.c:1350]: On Linux, realpath of self-loop does not seem to raise any errors.

**Review by mingweishih — APPROVED** [review]

**Review by mikbras — APPROVED** [review]
Really cool. Thank you for doing this!

**Review by vtikoo — APPROVED** [review]

**Pull request merged**